addons/account/models/account_journal_dashboard.py

```diff
@@ -702,44 +694,26 @@ def _get_draft_sales_purchases_query(self):
 
     def _get_open_sales_query(self):
         return self.env['account.move.line']._where_calc([
-            *self.env['account.move.line']._check_company_domain(self.env.companies),
-            ('journal_id', 'in', self.ids),
-            ('move_id.payment_state', 'in', ('not_paid', 'partial')),
+            ('move_id', 'in', self.env['account.move']._where_calc([
+                *self.env['account.move.line']._check_company_domain(self.env.companies),
+                ('journal_id', 'in', self.ids),
+                ('payment_state', 'in', ('not_paid', 'partial')),
+                ('move_type', '=', 'out_invoice'),
+                ('state', '=', 'posted'),
+            ])),
             ('account_type', '=', 'asset_receivable'),
-            ('move_id.move_type', '=', 'out_invoice'),
-            ('parent_state', '=', 'posted'),
-        ])
-
-    def _get_late_sales_query(self):
-        return self.env['account.move.line']._where_calc([
-            *self.env['account.move.line']._check_company_domain(self.env.companies),
-            ('journal_id', 'in', self.ids),
-            ('move_id.payment_state', 'in', ('not_paid', 'partial')),
-            ('account_type', '=', 'asset_receivable'),
-            ('date_maturity', '<', fields.Date.context_today(self)),
-            ('move_id.move_type', '=', 'out_invoice'),
-            ('parent_state', '=', 'posted'),
         ])
```

Comment:
Maybe we can merge the two methods to get a single main one since they look really similar now ?

```py
def _get_open_sales_query(self):
  return self._get_open_move_lines_query()

def _get_open_purchases_query(self):
  return self._get_open_move_lines_query("purchase")

def _get_open_move_lines_query(self, journal_type="sales"):
  return self.env['account.move.line']._where_calc([
    ('move_id', 'in', self.env['account.move']._where_calc([
      *self.env['account.move.line']._check_company_domain(self.env.companies),
      ('journal_id', 'in', self.ids),
      ('payment_state', 'in', ('not_paid', 'partial')),
      ('move_type', '=', 'in_invoice' if journal_type == 'purchase' else 'out_invoice'),
      ('state', '=', 'posted'),
    ])),
    ('account_type', '=', 'liability_payable' if journal_type == 'purchase' else 'asset_receivable'),
  ])
```